src/main/java/net/minecraftforge/fmp/block/BlockCoverable.java

```diff
@@ -0,0 +1,730 @@
+
+        List<AxisAlignedBB> list = new ArrayList<AxisAlignedBB>();
+        AxisAlignedBB box = entityBox.offset(-pos.getX(), -pos.getY(), -pos.getZ());
+        tile.getMicroblockContainer().getPartContainer().addCollisionBoxes(box, list, collidingEntity);
+        for (AxisAlignedBB aabb : list)
+        {
+            collidingBoxes.add(aabb.offset(pos));
+        }
+    }
+
+    public void addCollisionBoxToListDefault(IBlockState state, World worldIn, BlockPos pos, AxisAlignedBB entityBox,
+            List<AxisAlignedBB> collidingBoxes, Entity entityIn)
+    {
+        super.addCollisionBoxToList(state, worldIn, pos, entityBox, collidingBoxes, entityIn);
+    }
+
+    @Override
+    public final int getLightValue(IBlockState state, IBlockAccess world, BlockPos pos)
+    {
+        IMicroblockContainerTile tile = getMicroblockTile(world, pos);
+        return Math.max(tile != null ? tile.getMicroblockContainer().getPartContainer().getLightValue() : 0,
+                getLightValueDefault(state, world, pos));
+    }
+
+    public int getLightValueDefault(IBlockState state, IBlockAccess world, BlockPos pos)
+    {
+        return super.getLightValue(state, world, pos);
+    }
+
+    @Override
+    public final ItemStack getPickBlock(IBlockState state, RayTraceResult target, World world, BlockPos pos, EntityPlayer player)
+    {
+        if (target instanceof RayTraceResult)
+        {
+            IMicroblockContainerTile tile = getMicroblockTile(world, pos);
+            return tile != null ? tile.getMicroblockContainer().getPartContainer().getPickBlock(player, (RayTraceResult) target) : null;
+        }
+        return getPickBlockDefault(state, target, world, pos, player);
+    }
+
+    public ItemStack getPickBlockDefault(IBlockState state, RayTraceResult target, World world, BlockPos pos, EntityPlayer player)
+    {
+        return super.getPickBlock(state, target, world, pos, player);
+    }
+
+    private IMicroblockContainerTile brokenTile = null;
+    private boolean harvestingWrapper = false;
+
+    @Override
+    public final void harvestBlock(World world, EntityPlayer player, BlockPos pos, IBlockState state, TileEntity te, ItemStack stack)
+    {
+        brokenTile = te instanceof IMicroblockContainerTile ? (IMicroblockContainerTile) te : null;
+        harvestBlockDefault(world, player, pos, state, te, stack);
+        brokenTile = null;
+    }
+
+    public void harvestBlockDefault(World worldIn, EntityPlayer player, BlockPos pos, IBlockState state, TileEntity te, ItemStack stack)
+    {
+        super.harvestBlock(worldIn, player, pos, state, te, stack);
+    }
+
+    @Override
+    public final List<ItemStack> getDrops(IBlockAccess world, BlockPos pos, IBlockState state, int fortune)
+    {
+        if (harvestingWrapper) { return getDropsDefault(world, pos, state, fortune); }
+        IMicroblockContainerTile brokenTile = getMicroblockTile(world, pos);
+        if (brokenTile == null)
```

Comment:
Don't shade names like this on purpose... Also instance variables on Block is not a good idea..... Threads kill it...